src/service/users.js

```diff
@@ -237,3 +237,20 @@ export const checkFollowStatusBatch = async (targetUserIds = [], axiosOptions =
     );
   }
 };
+
+export const checkFollowStatus = async (channelId) => {
+  if (!channelId) return Promise.reject(new Error('Target User ID is empty'));
+
+  try {
+    const result = await api.get(`/users/check-follow-by-channelId?channelId=${channelId}`);
+    if (result?.status === 200) {
+      return Promise.resolve(result?.data?.data);
+    }
+
+    return Promise.reject(result?.data?.message);
+  } catch (e) {
+    return Promise.reject(
+      e?.response?.data?.message || "Can't check follow status, please try again later"
+    );
+  }
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `checkFollowStatus` function seems to be well implemented. It checks if the `channelId` is provided before making the API call, handles the response correctly by checking the status code and returns the appropriate data or error message. However, it's a good practice to encode URI components to ensure that special characters are correctly interpreted by the server.
```diff
-     const result = await api.get(`/users/check-follow-by-channelId?channelId=${channelId}`);
+     const result = await api.get(`/users/check-follow-by-channelId?channelId=${encodeURIComponent(channelId)}`);
```
Also, consider using constants for HTTP status codes for better readability and maintainability.
```diff
-     if (result?.status === 200) {
+     if (result?.status === HTTP_STATUS_OK) {
```
Remember to define `HTTP_STATUS_OK` as a constant with value `200`.


<!-- This is an auto-generated comment by OSS CodeRabbit -->